provisioning/service/src/Config/EnrollmentGroup.cs

```diff
@@ -258,14 +258,21 @@ public Attestation Attestation
             }
             set
             {
-                if(!((value ?? throw new ArgumentNullException(nameof(value))) is X509Attestation))
+                if (value == null)
                 {
-                    throw new ArgumentException("Attestation for enrollmentGroup shall be X509");
+                    throw new ArgumentNullException(nameof(value));
+                }
+                else if (!(value is X509Attestation) && !(value is SymmetricKeyAttestation))
+                {
+                    throw new ArgumentException("Attestation for enrollmentGroup shall be X509 or symmetric key");
                 }
 
-                if((((X509Attestation)value).RootCertificates == null) && (((X509Attestation)value).CAReferences == null))
+                if (value is X509Attestation)
                 {
-                    throw new ArgumentException("Attestation mechanism do not contains a valid certificate.");
+                    if ((((X509Attestation)value).RootCertificates == null) && (((X509Attestation)value).CAReferences == null))
+                    {
```

Comment:
Should we also check that Symmetric Key attestation primary and secondary keys are not null?

Reply:
It is optional for client to set these values. If not set the service will generate new keys during create

Reply:
we should let the svc valdiate

---
In reply to: [246905038](https://github.com/Azure/azure-iot-sdk-csharp/pull/750#discussion_r246905038) [](ancestors = 246905038)